docs/conf.py

```diff
@@ -92,6 +92,8 @@
 
 autodoc_typehints = 'description'
 
+autoclass_content = 'both'
```

Comment:
Are you sure we want this? This would impact all of our docstrings, are you sure it's better like that?

Reply:
It depends. The thing is that we aren't consistent (at least in the optim package but I expect it's the case elsewhere) in how we split between class and __init__ docstrings. We can not have this, but in some cases here we are missing some docs. Alternative is changing docstrings accordingly (can do for optim). Let me ask others